hyperspy/io_plugins/jeol.py

```diff
@@ -186,7 +192,54 @@ def read_img(filename, scale=None, **kwargs):
 
 def read_pts(filename, scale=None, rebin_energy=1, sum_frames=True,
              SI_dtype=np.uint8, cutoff_at_kV=None, downsample=1,
-             only_valid_data=True, **kwargs):
+             only_valid_data=True, read_em_image=False,
+             frame_list=[], frame_start_index=[], frame_shifts=[], 
+             lazy=False,
+             **kwargs):
+    """
+    rawdata : ndarray of uint16
+    	spectrum image part of pts file
+    scale : list of float
+    	-scale[2], scale[3] are the positional scale from asw data, 
+    	default is None, calc from pts internal data
+    rebin_energy : int
+    	Binning parameter along energy axis. Must be 2^n.
+    sum_frames : bool or ndarray of image shifts 
+	integrate along frame axis if sum_frames == True
+    SI_dtype : dtype
+        data type for spectrum image. default is uint8
+    cutoff_at_kV : float
+	cutoff energy to reduce memory size of spectrum image, default: None (do not cutoff)
+    downsample : int or [int, int]
+    	downsample along spacial axes to reduce memory size of spectrum image
+	value must be 2^n.
+	default: 1 (do not downsample)
+    only_valid_data : bool, default True
+    	read incomplete frame if only_valid_data == False
+        (usually interrupting mesurement makes incomplete frame)
+    read_em_image : bool, default False
+        read SEM/STEM image from pts file if read_em_image == True
+    frame_list : list of int [fr0, fr1, ...] or array [[fr0, dx0,dy0,dz0],...], default None
+    	list of frame numbers to be read (None for all data)
+    	if dx, dy, dz are specified, each images are shifted. dx, dy, dz are integers, in unit of pixel.
+	This is useful for express drift correction. (not suitable for a detailed analysis)
+    si_lazy : bool or tuple of chunksize, default False
+    	read spectrum image into sparse array if si_lazy == True
+    	SEM/STEM image is always read into dense array (np.ndarray)
+    	if tuple is specified, dask.array is generated using chunksize=si_lazy
+    lazy : bool, default False
+	set lazy flag not only spectrum image but also other data,
+	if lazy == True. This also set si_lazy = True.
+	Only the spectrum image data is read as a sparse array,
```

Comment:
Update since `si_lazy` has been removed and also mention sparse array - see other comments regarding lazy loading/sparse array.

Reply:
fixed.